modelling_opponent/PrisonerABC.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC, abstractmethod
```

Comment:
what is the purpose of this?

Reply:
As we discussed on Thursday, this is designed to organize the interface to all opponent types. I changed the method names and arguments per our discussion.